File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap <Integer, Integer> store = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];
            if (store.containsKey(diff)) {
                return new int[] {i, store.get(diff)};
            } else {
                store.put(nums[i], i);
            }
        }
        return new int[] {0,0};
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        store = {}
        for i in range(len(nums)):
            rem = target - nums[i]

            if rem in store:
                return [store[rem], i]
            else:
                store[nums[i]] = i